dora-parser/src/parser.rs

```diff
@@ -469,6 +472,19 @@ impl<'a> Parser<'a> {
         Ok(types)
     }
 
+    // Name is placeholder
+    fn parse_direct_trait_impls(&mut self) -> Result<Vec<Name>, ParseErrorAndPos> {
+        if self.token.is(TokenKind::Implements) {
+            self.advance_token()?;
+            let impls = self.parse_list_0(TokenKind::Comma, TokenKind::LBrace, false, |p| {
```

Comment:
Let's invoke another function here: parse_list_stop or so. parse_list and parse_list_stop would then use e.g. parse_list_inner internally.